### ISSUE TITLE
Task  dependencies / Linked tasks / Task Relationships

### ISSUE THREAD

**Issue by balporig** [super-productivity/super-productivity#2173]
## Problem Statement

There is no way to represent a connection between separate tasks (not the same as subtasks) from the same or different projects. One task can depend on another task, represent a different kind of relationship, like `is caused by`, `causes`, `relates to` etc.

## :grey_question: Possible Solution

A right click `Add relationship` option on selected task. It will bring a popup where you can input 1 or more tasks and the kind of relationships between them. In the additional info view, there can be a new `Linked tasks` section displaying the related tasks.

One more nice to have thing after this is implemented is referencing other notes by an ID in the `Description`, similar to how you can reference issues in github like this #15. Again, another option can be added to the context menu like `Copy task ID` and maybe be also displayed in the additional info section.

This will open the possibility in the future for a `Graph View` like the one by obsidian https://help.obsidian.md/Plugins/Graph+view or foam https://github.com/foambubble/foam to be developed.

**Comment by castarco** [comment]
I'd say that this task remains quite relevant, as the problem at hand has not been solved yet.

**Comment by OLLI-S** [comment]
There seem to be a few more open issues requesting the same or closely related features:

- #5214 asks for a "Related tasks" field with relationship types like *related*, *blocks/is blocked by*, and *follows/is followed by*
- #5148 asks specifically for dependent tasks (a task only becomes visible once the task it depends on is completed)
- #4790 asks for dependent tasks plus a Gantt chart visualization (proposed as a plugin)

All four issues share the same root need: representing relationships between tasks. It might make sense to consolidate the discussion here or in #5214.

**Comment by robertoost** [comment]
Thank you for opening this issue. I also think this would be a great feature. I'm a new SP user so I was confused when I couldn't find this feature. Let me sketch the situation that made me wonder whether I'd need this functionality: 

I have the scrum master role in our team. During the sprint planning, we notice a bug that has to be solved by Thursday. We assign a team member, and I have to make sure it gets finished. I create a reminder task: If the person assigned can't finish by tuesday, I transfer the ticket to our senior team member.

Assigning myself to the ticket in Gitlab just for a reminder doesn't fit in our workflow, so I use the Gitlab integration to link the task to Super Productivity. I create the task: "check in with teammember about feature." I now want to link the feature to that reminder task, but it doesn't give me the option. Changing the gitlab-sourced task's title isn't an option, two-way sync is enabled.

Instead, I remove the gitlab item and manually paste the url to the gitlab ticket in the url section of the reminder task.

So yeah, all in all there's opportunity here for a nicer workflow. If anyone thinks there's a better way for me to solve this situation then I'm all ears. Even better if the solution is already a feature within SP I could use.

**Comment by ramisedhom** [comment]
I would like to see this feature in SP soon. I think the first step is allowing SP to have deep link, something like: super-productivity://open-task?id=<task_id>. Should we open a separate request for deep link first?
